File: codewars/isValidIP.py
def is_valid_IP(strng = False):
    if strng == '' or strng == None:
        return False
    elif strng == False:
        return False
    elif strng.count('.') == 3:
        for i in strng.split('.'):
            if i.isdigit() and 0 <= int(i) <= 255 and (len(i) == 1 or len(i) > 1 and i[0] != '0'):
                continue
            else:
                return False
        return True
    return None

print(is_valid_IP())

File: codewars/isValidIP.js
/*
identify valid IPv4 addresses in dot-decimal format. 
IPs should be considered valid if they consist of four octets, 
with values between 0 and 255, inclusive.
*/

function isValidIP(str) {
    if (typeof str !== 'string'){ return false}
    else if (str === ''){ return false}
    else {
        let arr = str.split('.')
        if (arr.length !== 4) { return false}
        for (let i = 0;i < arr.length;i++) {
            if (arr[i] === '') { return false}
            else if (0 <= arr[i] <= 255) { return false}
            else if (arr[i].match(/[^0-9]/g)) { return false}
        }
    }
    
    return true
}

console.log('01. 02.03'.split('.'))
